main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DuplicateRouteNameValidator.java

```diff
@@ -48,56 +53,89 @@ public void validate(NoticeContainer noticeContainer) {
                   }
                   return;
                 } else {
-                  routeByLongName.put(route.routeLongName(), route);
+                  routeByLongName.put(getRouteLongNameKey(route), route);
                 }
               }
               if (route.hasRouteShortName()) {
-                if (routeByShortName.containsKey(route.routeShortName())) {
+                if (routeByShortName.containsKey(getRouteShortNameKey(route))) {
                   if (areRoutesFromSameAgency(
-                      route.agencyId(), routeByShortName.get(route.routeShortName()).agencyId())) {
+                      route.agencyId(),
+                      routeByShortName
+                          .get(getRouteShortNameKey(route))
+                          .agencyId())) {
                     noticeContainer.addValidationNotice(
                         new DuplicateRouteNameNotice(
                             "route_short_name", route.csvRowNumber(), route.routeId()));
                   }
                 } else {
-                  routeByShortName.put(route.routeShortName(), route);
+                  routeByShortName.put(getRouteShortNameKey(route), route);
                 }
               }
             });
   }
 
   /**
-   * Determines if two routes are from the same agency
+   * Determines if two routes are from the same agency: ids are case-sensitive.
    *
    * @param routeAgencyId first agency_id
    * @param otherRouteAgencyId second agency_id
    * @return true if both agency ids are equals returns false otherwise.
    */
   private boolean areRoutesFromSameAgency(
       final String routeAgencyId, final String otherRouteAgencyId) {
-    return routeAgencyId.equalsIgnoreCase(otherRouteAgencyId);
+    return routeAgencyId.equals(otherRouteAgencyId);
+  }
+
+  /**
+   * Generate a key used to store {@code GtfsRoute} by `routes.route_long_name`.
+   *
+   * @param route the {@code GtfsRoute} to generate the key from
+   * @return `routes.route_long_name`+`route.routeType`
+   */
+  private String getRouteLongNameKey(GtfsRoute route) {
+    return route.routeLongName() + route.routeType();
+  }
+
+  /**
+   * Generate a key used to store {@code GtfsRoute} by `routes.route_short_name`.
+   *
+   * @param route the {@code GtfsRoute} to generate the key from
+   * @return `routes.route_short_name`+`route.routeType`
+   */
+  private String getRouteShortNameKey(GtfsRoute route) {
+    return route.routeShortName() + route.routeType();
+  }
+
+  /**
+   * Generate a key used to store {@code GtfsRoute} by both `routes.route_short_name` and
+   * `routes.route_long_name`.
+   *
+   * @param route the {@code GtfsRoute} to generate the key from
+   * @return `routes.route_short_name`+`routes.route_long_name`+`route.routeType`
+   */
+  private String getShortAndLongNameKey(GtfsRoute route) {
+    return route.routeShortName() + route.routeLongName() + route.routeType();
```

Comment:
Concatenation of strings is a non-efficient operation that leads to memory fragmentation. A hashcode would suffice here.

Reply:
Agreed, these could be replaced with `Objects.hash(...)`

Reply:
Thanks for the review. The changes will be linked to https://github.com/MobilityData/gtfs-validator/issues/805.

Reply:
Thanks for flagging that, the changes have been provided in #805 and #812